pubsub/awspubsub/awspubsub.go

```diff
@@ -72,41 +72,71 @@ func openTopic(ctx context.Context, client *sns.SNS, topicARN string) driver.Top
 var stringDataType = aws.String("String")
 
 // SendBatch implements driver.Topic.SendBatch.
-func (t *topic) SendBatch(ctx context.Context, dms []*driver.Message) (er error) {
+func (t *topic) SendBatch(ctx context.Context, dms []*driver.Message) error {
+	// Limit our concurrent RPCs via a semaphore.
+	const maxConcurrency = 10
+	sem := make(chan bool, maxConcurrency)
```

Comment:
An alternative would be allocate `sem` as a field of t. This would be better at limiting overall concurrent RPCs to AWS SNS (since `SendBatch` can be called concurrently, this PR as written only limits to 10x something).

I'd then have to use `errc` as the way to wait for all goroutines to complete, but that should be feasible (can't break out of the `for` loop on `ctx.Done`, and would have to read all `len(dms)` errors from `errc`).

WDYT?

Reply:
Seems reasonable, but what you have here is a good starting point. How about doing that in a later PR?

Reply:
If you use chan struct{}, the channel will take constant space. Not a big deal here, but worth getting into the habit.

Reply:
Done on `chan struct{}`.

@jba WDYT about the alternative described above?

Reply:
Not sure why you can't break out of the loop on ctx.Done. Can't you just count the number of goroutines you've launched?

Reply:
I will tackle this in a separate PR.